src/main/java/com/hillel/dzhurynskyi/homework/hm16/Tree.java

```diff
@@ -0,0 +1,69 @@
+package com.hillel.dzhurynskyi.homework.hm16;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Stack;
+
+public class Tree {
+    private Node root;
+
+    public Tree() {
+        this.root = null;
+    }
+
+    public void add(int value) {
+        Node newNode = new Node();
+        newNode.setValue(value);
+        if (root == null) {
+            root = newNode;
+        } else {
+            Node currentNode = root;
+            Node parentNode;
+            while (true) {
+                parentNode = currentNode;
+                if (value == currentNode.getValue()) {
+                    return;
+                } else if (value < currentNode.getValue()) {
+                    currentNode = currentNode.getLeftChild();
+                    if (currentNode == null) {
+                        parentNode.setLeftChild(newNode);
+                        return;
+                    }
+                } else {
+                    currentNode = currentNode.getRightChild();
+                    if (currentNode == null) {
+                        parentNode.setRightChild(newNode);
+                        return;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
consider a recursive method to do the add operation, it might be interesting for further implementations